Sources/apm-agent-ios/Instrumentation/CrashReporting/CrashManager.swift

```diff
@@ -0,0 +1,126 @@
+    let group : EventLoopGroup
+    let loggerProvider : LoggerProvider
+    init(resource: Resource, group: EventLoopGroup, agentConfiguration: AgentConfiguration) {
+        self.group = group
+        // if something went wrong with the lastResource in the user defaults, fallback of the current resource data.
+        var tempResource = resource
+        
+        let otlpConfiguration = OtlpConfiguration(timeout: OtlpConfiguration.DefaultTimeoutInterval, headers: OpenTelemetryHelper.generateExporterHeaders(agentConfiguration.auth))
+
+        
+        if let lastResourceJson  = UserDefaults.standard.data(forKey: Self.lastResourceDefaultsKey) {
+            do {
+                let decoder = JSONDecoder()
+                tempResource = try decoder.decode(Resource.self, from: lastResourceJson)
+            } catch {
+                os_log("[Elastic][CrashManager] initialization: unable to load last Resource from user defaults.")
+            }
+        }
+        lastResource = tempResource
+        loggerProvider = LoggerProviderBuilder()
+            .with(resource: lastResource)
+            .with(processors: [SimpleLogRecordProcessor(logRecordExporter: OtlpLogExporter(channel: OpenTelemetryHelper.getChannel(with: agentConfiguration, group: group),
+                                                                                           config: otlpConfiguration,
+                                                                                           logger: Logger(label:Self.LOG_LABEL), envVarHeaders: OpenTelemetryHelper.generateExporterHeaders(agentConfiguration.auth)))])
+            .build()
+        do {
+            let encoder = JSONEncoder()
+            let data = try encoder.encode(resource)
+            UserDefaults.standard.set(data, forKey:Self.lastResourceDefaultsKey)
+        } catch {
+            os_log("[Elastic][CrashManager] initialization: unable to save current Resource from user defaults.")
+
+        }
+    }
+    
+    public func initializeCrashReporter() {
+        // It is strongly recommended that local symbolication only be enabled for non-release builds.
+        // Use [] for release versions.
+        let config = PLCrashReporterConfig(signalHandlerType: .mach, symbolicationStrategy: [])
+        guard let crashReporter = PLCrashReporter(configuration: config) else {
+          print("Could not create an instance of PLCrashReporter")
+          return
+        }
+
+        // Enable the Crash Reporter.
+        do {
+          try crashReporter.enableAndReturnError()
+        } catch let error {
+          print("Warning: Could not enable crash reporter: \(error)")
+        }
+        
+        // Try loading the crash report.
+        if crashReporter.hasPendingCrashReport() {
+          do {
+            let data = try crashReporter.loadPendingCrashReportDataAndReturnError()
+              let lp = loggerProvider.loggerBuilder(instrumentationScopeName: "CrashReport")
```

Comment:
Do you think this scope name should be the same on both platforms? If so, I think it's worth adding it [here](https://github.com/elastic/apm/pull/728/files).

Reply:
I don't think so. This name represents the instrumentation package, it's not related to the data collected. We've not considered this for any other instrumentation packages. I might update this to "PLCrashReporter" to better represent what this instrumentation is.

Reply:
Go it, sounds good